### ISSUE TITLE
Can't manage to make it work

### ISSUE THREAD

**Issue by zreedeecom** [deliciousbrains/wp-migrations#4]
Hi, after some hours trying to make this thing work I think I'm missing something..

I followed the readme steps with no luck, I also tried to filer the path and even checking the filter got applied I always get the same message `Warning: There are no migrations to run.`

I'm requiring wp-migrations from the root of my project as I'm also pulling wp and other deps via composer.

Not sure what you mean on the readme with "Bootstrap the package by adding \DeliciousBrains\WPMigrations\Database\Migrator::instance(); to an mu-plugin."

I got that line on a mu-plugin but still no luck. I got another error also when trying to run wp dbi migrate, I don't remember it but after running `wp dbi migrate --setup` it went away leaving me with the mentioned one.

I also tried using the default route `app/migrations/AddTestTable.php` (removing my filter of course) and same message is printed out.

Any hints?? Thanks!

**Comment by polevaultweb** [comment]
What does your `AddTestTable.php` file look like?

**Comment by zreedeecom** [comment]
Nothing special.. just wanted to test if it works and then do a rollback.

```php
<?php

# app/migrations/AddTestTable.php

use DeliciousBrains\WPMigrations\Database\AbstractMigration;

class AddTestTable extends AbstractMigration
{

  public function run()
  {
    global $wpdb;

    $sql = "
            CREATE TABLE " . $wpdb->prefix . "test_table (
            id bigint(20) NOT NULL auto_increment,
            some_column varchar(50) NOT NULL,
            PRIMARY KEY (id)
            ) {$this->get_collation()};
        ";

    dbDelta($sql);
  }

  public function rollback()
  {
    global $wpdb;
    $wpdb->query('DROP TABLE ' . $wpdb->prefix . 'test_table');
  }
}
```

Project structure looks like this https://cl.ly/e01fa4d06201

**Comment by polevaultweb** [comment]
Can you commit what you have?

**Comment by zreedeecom** [comment]
🤦‍♂ sorry, I thought it was pushed, check again please https://github.com/zreedeecom/wordpress-skeleton/tree/add-wp-migrations-support

**Comment by polevaultweb** [comment]
Apologies, there is an assumption made about the migration filename. It needs to be in the format `yyyy_mm_dd_classname`, eg. `2019_10_16_AddTestTable.php`

**Comment by zreedeecom** [comment]
Oh I see, I'll test, not a big deal to add the date tho..
Just out of curiosity.. where is that written (in the code)? because I read your code and didn't find anything about that, as far as I see it's 1. finding out the migrations folder/s 2.listing all php files on that folder and 3. running all migrations without a record on migrations table.

I like a lot the idea of using this laravel flavored kind of db migrations on my skeleton project but I want to document it well, so others can follow just fine without having to dig as much as I'm doing to fit all these pieces together.

**Comment by zreedeecom** [comment]
I've been testing it and these are the results:

- It's needed to manually run `wp dbi migrate --setup` , it won't be fired automatically for new installs. Maybe a quick check to see if `dbrns_migrations`table exists and if it doesn't trigger setup when user tries to migrate for first time? or at least add it on the readme, may I do it for you?
- Renamed the file as you stated and it works 👍 from default location `/app/migrations`
- [Applied the filter from the mu-plugin](https://github.com/zreedeecom/wordpress-skeleton/blob/fcf8f5c8e7cb795c1fda79fae9d4cbdb6d3b7dc0/public/wp-content/mu-plugins/wp_migrations.php#L37) and it works fine from the new location.
- I can't rollback the migration, tried: `wp dbi migrate --rollback``wp dbi migrate 2019_10_16_AddTestTable.php --rollback` `wp dbi migrate 2019_10_16_AddTestTable.php --rollback=true`

**Comment by wfinn2** [comment]
Hey I was having the same issues as @zreedeecom and when i run `wp dbi migrate --setup` i keep getting `Error: Error establishing a database connection.` I am running the site on local and all of the DB credentials are correct (I know this because the site is operating while I'm trying to get this to work). Any suggestions?

**Comment by polevaultweb** [comment]
@wfinn2 can you run a WP CLI command successfully without a database connection issue? eg. `wp plugin install akismet`

**Comment by wfinn2** [comment]
@polevaultweb Yes, wp-cli commands work with out a database connection error. I restarted the entire process by deleting all associated files. I then:
1. ran `composer require deliciousbrains/wp-migrations` 
2. I copied the wp-migrations.php file into my mu-plugin directory that @zreedeecom has provided in his "wordpress-skeleton" in order the complete the second step of your installation instructions (because i don't fully understand what this instruction means). Now my site crashed saying `Error: Class 'DeliciousBrains\WPMigrations\Database\Migrator' not found` What have I done?

If you need more details please let me know

Thank you for your timely response and any future help you can provide!

**Comment by achatainga** [comment]
I am exactly at the same point of @zreedeecom 
I am using https://roots.io/bedrock/ I have created a mu-plugin and installed it.
I created ./app/migrations/2021_08_04_testMigration.php
`wp dbi migrate` works
`wp dbi migrate --rollback` does not work
I am using `./vendor/bin/wp` and it does work well to install update or do whatever wp-cli command.

Rollback is not working on my setup. Does it work on anyother setup?

Thanks in advanced
